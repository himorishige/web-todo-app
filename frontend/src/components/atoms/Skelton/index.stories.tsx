/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Skelton from '.';

export default {
  title: 'WebToDoApp/Atoms/Skelton',
  component: Skelton,
} as ComponentMeta<typeof Skelton>;

const Template: ComponentStory<typeof Skelton> = (args) => <Skelton {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Skelton';
Default.args = {
  width: '100%',
  height: '1.5rem',
  radius: '0',
};
