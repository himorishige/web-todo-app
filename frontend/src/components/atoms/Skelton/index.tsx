/** @jsxImportSource @emotion/react */
import React from 'react';
import { css, keyframes } from '@emotion/react';

export type Props = {
  height?: string;
  width?: string;
  radius?: string;
};

const Skelton: React.VFC<Props> = (props) => {
  const { height = '4rem', width = '100%', radius = '0' } = props;

  return (
    <div
      css={skeltonStyle}
      style={{ width: `${width}`, height: `${height}`, borderRadius: `${radius}` }}
    />
  );
};

export default Skelton;

const skeltonAnimation = keyframes`
  0% {
    transform: translateX(-100%)
  }
  100% {
    transform: translateX(100%)
  }
`;

const skeltonStyle = css`
  width: 100%;
  height: 4rem;
  line-height: 2;
  background: rgba(206, 229, 208, 1);
  position: relative;
  overflow: hidden;
  margin: 0;
  &::before {
    content: '';
    display: block;
    height: 100%;
    width: 100%;
    background: linear-gradient(90deg, transparent, rgba(218, 218, 218, 0.4), transparent);
    position: absolute;
    top: 0;
    left: 0;
    z-index: 100;
    animation: ${skeltonAnimation} 1.2s linear infinite;
  }
`;
