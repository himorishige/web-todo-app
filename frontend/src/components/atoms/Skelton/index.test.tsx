import renderer from 'react-test-renderer';
import Skelton from '.';

describe('Atoms/Skelton', () => {
  it('正しくレンダリングされている', () => {
    const element = renderer.create(<Skelton />).toJSON();
    expect(element).toMatchSnapshot();
  });
});
