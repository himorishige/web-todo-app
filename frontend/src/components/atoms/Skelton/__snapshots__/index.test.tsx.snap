// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Skelton 正しくレンダリングされている 1`] = `
<div
  className="css-1976yk9"
  style={
    Object {
      "borderRadius": "0",
      "height": "4rem",
      "width": "100%",
    }
  }
/>
`;
